sql/core/pom.xml

```diff
@@ -147,6 +147,11 @@
       <groupId>org.apache.xbean</groupId>
       <artifactId>xbean-asm9-shaded</artifactId>
     </dependency>
+    <dependency>
```

Comment:
This is not required, right?

Reply:
For sbt, it is not required:

```
[info] KVStoreProtobufSerializerSuite:
[info] - SQLExecutionUIData (121 milliseconds)
[info] Run completed in 1 second, 648 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 293 s (04:53), completed 2022-12-23 9:41:14
```

**But for maven, it is required now**:

```
mvn clean test -pl sql/core -am -Dtest=none -DwildcardSuites=org.apache.spark.status.protobuf.sql.KVStoreProtobufSerializerSuite
```

```
[INFO] --- scala-maven-plugin:4.8.0:compile (scala-compile-first) @ spark-sql_2.12 ---
[INFO] Compiler bridge file: /Users/yangjie01/.sbt/1.0/zinc/org.scala-sbt/org.scala-sbt-compiler-bridge_2.12-1.8.0-bin_2.12.17__52.0-1.8.0_20221110T195421.jar
[INFO] compiler plugin: BasicArtifact(com.github.ghik,silencer-plugin_2.12.17,1.7.10,null)
[INFO] compiling 576 Scala sources and 75 Java sources to /spark-source/sql/core/target/scala-2.12/classes ...
[ERROR] [Error] /spark-source/sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/SQLExecutionUIDataSerializer.scala:35: Class com.google.protobuf.GeneratedMessageV3 not found - continuing with a stub.
[ERROR] [Error] : Unable to locate class corresponding to inner class entry for Builder in owner com.google.protobuf.GeneratedMessageV3
[ERROR] [Error] /spark-source/sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/SQLExecutionUIDataSerializer.scala:43: Class com.google.protobuf.GeneratedMessageV3 not found - continuing with a stub.
[ERROR] [Error] /spark-source/sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/SQLExecutionUIDataSerializer.scala:55: value toByteArray is not a member of org.apache.spark.status.protobuf.StoreTypes.SQLExecutionUIData
[ERROR] [Error] : Unable to locate class corresponding to inner class entry for Builder in owner com.google.protobuf.GeneratedMessageV3
[ERROR] 5 errors found
```

during to sql module inherits `protobuf-java` 2.5.0 with compile scope from parent `pom.xml`. We can verify this from the [sql-module-dependency-tree.txt](https://github.com/apache/spark/files/10291491/sql-module-dependency-tree.txt)

Reply:
https://github.com/apache/spark/pull/39164#discussion_r1055263888

If the place we mentioned yesterday can be changed to 3.21.11, then we can remove this declaration

Reply:
ok, do we need to shade it?

Reply:
Don't need shaded, I make a spark client with this pr for testing, and it can be parsed normally

Reply:
Hmmmm, did you test with both Maven and SBT? Then why core module requires shading..

Reply:
Yes, both MAVEN and SBT tested. 

Personally, I think the core module needs shaded+relocation because we don't want protobuf version Spark depends on to affect other third-party projects, just like `connect`&`protobuf` module.  And the SQL module only uses the protobuf class generated in the core module, so it does not need to shaded + relocate again.

In fact, because Spark already uses the unified protobuf version, the `connect` and `protobuf` module only need relocation protobuf package to keep consistent with the relocation rules of the core module , rather than the shaded `protobuf-java` again.